### PR TITLE
Contributing: Scan Rules

### PR DESCRIPTION
**Pull request by psiinon**
Anything else that could or should be mentioned?

Screenshot 2022-03-04 at 15-49-58 OWASP ZAP – Contribute Scan Rules

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: site/content/docs/contribute/scan-rules.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+#### Name
+
+Alert names must not be dynamic.
+
+A rule can raise alerts with differing names but it must then implement the [getExampleAlerts()](https://javadoc.io/doc/org.zaproxy/zap/latest/org/zaproxy/zap/extension/alert/ExampleAlertProvider.html) method (which is used to populate the [alert pages](/docs/alerts/)) and [set the AlertRef](https://javadoc.io/static/org.zaproxy/zap/2.11.1/org/parosproxy/paros/core/scanner/Alert.html#setAlertRef-java.lang.String-) field correctly.
+
+#### Parameter
+
+The name of the url or form parameter or the header being attacked or affected. It must be left empty if none are relevant.
```
Comment (kingthorin) [inline site/content/docs/contribute/scan-rules.md:83]:
```suggestion
The name of the URL or form parameter or the header being attacked or affected. It must be left empty if none are relevant.
```

Comment (psiinon) [reply site/content/docs/contribute/scan-rules.md]: Good spot - fixed.

File: site/content/docs/contribute/scan-rules.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+Active scan rules should limit the number of requests they make based on the Strength set by the user:
+
+* LOW:      Limit to around 6 requests per scan call
+* MEDIUM:   Limit to around 12 requests
+* HIGH:     Limit to around 24 requests
+* INSANE:   No limit, although 1000s wouldn’t be a good idea
+
+### Alert Field Details
+
+The following advice and guidance should be followed for any new alerts raised.
```
Comment (ricekot) [inline site/content/docs/contribute/scan-rules.md:67]: A link to https://www.zaproxy.org/docs/desktop/ui/dialogs/addalert/ can be added here. It contains some info on other fields like Risk, Confidence and Alert Tags which are not mentioned here.

Comment (psiinon) [reply site/content/docs/contribute/scan-rules.md:67]: Done :)

**Review by ricekot — APPROVED** [review]
Thanks!

File: site/content/docs/contribute/scan-rules.md (hunk 2 above)
Comment (ricekot) [reply site/content/docs/contribute/scan-rules.md:67]: Thanks!

**Pull request merged**